mountpoint-s3/src/fs.rs

```diff
@@ -992,12 +1000,21 @@ where
                 .ok_or_else(|| err!(libc::EBADF, "invalid directory handle"))?
         };
 
+        // special case where we need to rewind and restart the streaming
+        if offset == 0 && dir_handle.offset() != 0 {
+            // only do this if this is not the first call with offset 0
+            dir_handle.rewind_offset();
+            let new_handle = self.default_handle(parent).await?;
```

Comment:
I think we need to `rewind_offset` _after_ creating the new handle — if handle creation fails we want to leave the old one in place (which `?` is doing here) and the offset unchanged.